File: 0226-invert-binary-tree/0226-invert-binary-tree.py
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        queue=deque()#using a deque is good for append,pop,popleft, and appendleft
        queue.append(root)#initially add the root
        while queue:#until this is empty
            currentNode=queue.popleft()#take out the first element
            if currentNode:
                currentNode.left,currentNode.right=currentNode.right,currentNode.left #do the swapping in single operation as otherwise the values of left or right change.
            
                if currentNode.left:
                    queue.append(currentNode.left)#add the subtrees of the nodes as well as they need to be swapped as well.
                if currentNode.right:
                    queue.append(currentNode.right)
        return root

File: 0226-invert-binary-tree/0226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if(root==null){
            return root;
        }
        ArrayDeque<TreeNode> deque=new ArrayDeque<>();
        deque.offer(root);
        while(deque.size()!=0){
            TreeNode currentNode=deque.poll();
            if(currentNode!=null){
                TreeNode temp=currentNode.left;
                currentNode.left=currentNode.right;
                currentNode.right=temp;
                if(currentNode.left!=null){
                    deque.offer(currentNode.left);
                }
                if(currentNode.right!=null){
                    deque.offer(currentNode.right);
                }
                
            }
        }
        return root;
        
        
    }
}